src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,264 @@
+import { UIText } from 'src/ui/ui-kit/UIText';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import type { Networks } from 'src/modules/networks/Networks';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { UnstyledButton } from 'src/ui/ui-kit/UnstyledButton';
+import { HStack } from 'src/ui/ui-kit/HStack';
+import { useNetworks } from 'src/modules/networks/useNetworks';
+import { getFungibleAsset } from 'src/modules/ethereum/transactions/actionAsset';
+import BigNumber from 'bignumber.js';
+import { formatCurrencyValue } from 'src/shared/units/formatCurrencyValue';
+import { minus, noValueDash } from 'src/ui/shared/typography';
+import { formatTokenValue } from 'src/shared/units/formatTokenValue';
+import { getDecimals } from 'src/modules/networks/asset';
+import { baseToCommon } from 'src/shared/units/convert';
+import { createChain } from 'src/modules/networks/Chain';
+import { CircleSpinner } from 'src/ui/ui-kit/CircleSpinner';
+import { formatPriceValue } from 'src/shared/units/formatPriceValue';
+import { PageFullBleedColumn } from 'src/ui/components/PageFullBleedColumn';
+import { ActionDetailedView } from '../History/ActionDetailedView';
+import * as styles from './styles.module.css';
+
+const dateFormatter = new Intl.DateTimeFormat('en', {
+  year: 'numeric',
+  month: 'short',
+  day: 'numeric',
+  hour: '2-digit',
+  minute: '2-digit',
+});
+
+function AssetHistoryItem({
+  action,
+  asset,
+  address,
+  networks,
+}: {
+  action: AddressAction;
+  asset: Asset;
+  address: string;
+  networks: Networks;
+}) {
+  const { currency } = useCurrency();
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+
+  const handleDialogOpen = useCallback(() => {
+    dialogRef.current?.showModal();
+  }, []);
+
+  const { transfer, isIncoming } = useMemo(() => {
+    const incomingTransfers = action.content?.transfers?.incoming?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedIncomingTransfer: ActionTransfer | undefined =
+      incomingTransfers?.length
+        ? {
+            ...incomingTransfers[0],
+            quantity: incomingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    const outgoingTransfers = action.content?.transfers?.outgoing?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedOutgoingTransfer: ActionTransfer | undefined =
+      outgoingTransfers?.length
+        ? {
+            ...outgoingTransfers[0],
+            quantity: outgoingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    return {
+      transfer: aggregatedIncomingTransfer || aggregatedOutgoingTransfer,
+      isIncoming: Boolean(aggregatedIncomingTransfer),
```

Comment:
isn't this same as 
```suggestion
      isIncoming: Boolean(incomingTransfers?.length),
```
?

Reply:
Yes, this is also ok. Not sure that this is much easier, but I can change it

Reply:
I mean, it doesn't rely on a derived value, so if you later remove `aggregatedIncomingTransfer`, you will still be able to keep `isIncoming`